runtime/tests/interpreter/reference_test.go

```diff
@@ -1815,3 +1816,1073 @@ func TestInterpretReferenceToReference(t *testing.T) {
 		require.ErrorAs(t, err, &interpreter.NestedReferenceError{})
 	})
 }
+
+func TestInterpretReferenceDereference(t *testing.T) {
```

Comment:
Can you add a test for optional references using optional chaining? I.e. something of type `&Int?` being dereferenced using `ref?.dereference()`?

Reply:
Done in 43d998901